src/applications/Machine.java

```diff
@@ -54,4 +54,8 @@ public void setActiveJob(Job activeJob) {
     public boolean isInactive() {
         return (activeJob == null);
     }
+
+    public boolean noWaitingJob() {
```

Comment:
This is more nit picky for me personally so feel free to ignore if you guys do not feel it is a worth while change:
The name `noWaitingJob()` seems a little clunky I feel a name like `noJobQueued()` would be a more clearer name for this function

Reply:
Nice point!! noJobQueued makes me think of 'no jobs in queue' rather than 'no job is currently waiting'. The first thought fit the scenario more as LinkedQueue are used to store Jobs.

Reply:
@UtkarshKr007 
I don't think having a method reflect its implementation is the best idea, especially in this case since the method is being called by things that are from another class. 
I do feel like the name is a bit better, but I don't entirely agree upon the reasoning.